### PR TITLE
Quality of life changes

### PR DESCRIPTION
**Pull request by CornholioNeedsRolio**
This is a version of baritone I made with what I needed for my projects, but after I implemented #homearea I felt like this really needed to be in the master. If there are bad practices I'm sure that they can fixed fairly easy.

### CODE REVIEW

File: src/main/java/baritone/cache/CachedHomeAreas.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                Files.createDirectories(directory);
+            } catch (IOException ignored) {
+            }
+        }
+
+    }
+    @Override
+    public synchronized void save(ISelectionManager manager)
+    {
+        Path fileName = this.directory.resolve("CoolShit.mp4");
```
Comment (scorbett123) [inline src/main/java/baritone/cache/CachedHomeAreas.java:46]: Don't swear where it is seen by end users, it is fine in comments, but preferably not creating the file on their computer.

File: src/main/java/baritone/command/defaults/HauseArea.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+        public static String[] getAllNames() {
+            Set<String> names = new HashSet<>();
+            for (TransformTarget target : TransformTarget.values()) {
+                names.addAll(Arrays.asList(target.names));
+            }
+            return names.toArray(new String[0]);
+        }
+    }
+}
```
Comment (scorbett123) [inline src/main/java/baritone/command/defaults/HauseArea.java:265]: Can the existing sel command be reused?

File: src/main/java/baritone/selection/SelectionManager.java [hunk 3]
```diff
@@ -13,19 +18,28 @@
...
+    public SelectionManager(Baritone baritone, boolean homeAreaManager) {
         new SelectionRenderer(baritone, this);
+        isHomeArea = homeAreaManager;
+        this.baritone = baritone;
     }
 
     private void resetSelectionsArr() {
         selectionsArr = selections.toArray(new ISelection[0]);
+        if(isHomeArea)
+            baritone.getWorldProvider().getCurrentWorld().getCachedHomeAreas().save(this);
```
Comment (scorbett123) [inline src/main/java/baritone/selection/SelectionManager.java:33]: formatting always use {}

**Review by scorbett123 — CHANGES_REQUESTED** [review]
Is it possible that these could be made as multiple pull requests, there are some changes that I think we want, but others that seem to specialised to map art that would make it to complicated for everyday users. 

The comments I have made are just a very quick look through, if you separate this into separate pull requests I will go through in more detail

File: src/api/java/baritone/api/Settings.java [hunk 4]
```diff
@@ -44,6 +44,32 @@
  */
 public final class Settings {
 
+    /**
+     * will not place or break any block
+     */
+    public final Setting<Boolean> safeFarming = new Setting<>(false);
```
[the remark this answers is not in the document]
Comment (CornholioNeedsRolio) [reply src/api/java/baritone/api/Settings.java:50]: it was getting tedious to keep doing that, I'm really bad at micromanagement

File: src/main/java/baritone/command/defaults/HauseArea.java (hunk 2 above)
Comment (CornholioNeedsRolio) [reply src/main/java/baritone/command/defaults/HauseArea.java:265]: i was planning doing it today, i just wanted to get it running

File: README.md [hunk 5]
```diff
@@ -1,121 +1,19 @@
...
+	-buildInLayersSidewaysXAxis: it's in direct connection with the buildInLayersSideways setting, if this is false it will build on z axis, otherwise on the x axis
+	
+	-dontPlaceBlocksThatAreNotFull: this option should make baritone not place incomplete blocks(carpets, cobwebs, slabs, etc), to avoid a pesky bug(where it gets stuck in an infinite loop of placing and breaking), place them yourself, I had a fix to avoid it getting in infinite loops but it was just disgusting, I have to familiarize even more with bartione's source code to fix that, until then again place them yourself
+    -mapArtMode: not new, but I changed how it works, now it will care about any blocks above the first non-air block, including the original block. Usefull if you have some blocks that need a block under them to be placed like carpets
+    -homearea(or ha for short): is a cool feature, it works like sel but whatever is inside this selection will be protected from baritone, no longer it will break your walls or place blocks in the middle of your living room, now it will use the door as a normal being :P
+    -safeFarming: baritone wont break or place blocks while #farm -ing
```
[the remark this answers is not in the document]
Comment (CornholioNeedsRolio) [reply README.md:19]: thanks for all the feedback you gave me! I will consider it

**Pull request closed without merging**